gwcs/wcs.py

```diff
@@ -1432,13 +1436,17 @@ def to_fits_sip(self, bounding_box=None, max_pix_error=0.25, degree=None,
             bounding_box = self.bounding_box
 
         (xmin, xmax), (ymin, ymax) = bounding_box
-        crpix1 = (xmax - xmin) // 2
-        crpix2 = (ymax - ymin) // 2
+        if crpix is None:
+            crpix1 = int(_py2round((xmax + xmin) / 2))
+            crpix2 = int(_py2round((ymax + ymin) / 2))
```

Comment:
Is there a strong reason to round this and cast to integer?  Generally CRPIXn is a float.

Reply:
@jdavies-st In general, CRPIX is a `float` but if _we_ have to guess one value, one not choose an easily readable one?

With regard to rounding, it is due making thing s in agreement with documentation: "closest pixel to the center of the bounding box". `ceil` or `floor` are not closest, sometimes.